lib/mods_display/fields/imprint.rb

```diff
@@ -42,167 +42,175 @@ def dates(element)
+      # Decoded dates with "B.C." or "A.D." and qualifier markers applied to
+      # the entire range, or individually if dates differ.
+      def qualified_value
+        if @start&.qualifier != @stop&.qualifier
+          "#{@start&.qualified_value}-#{@stop&.qualified_value}"
         else
-          date_field.content
+          qualifier = @start&.qualifier || @stop&.qualifier
+          date = "#{@start&.bc_ad_value}-#{@stop&.bc_ad_value}"
+          return "[ca. #{date}]" if qualifier == 'approximate'
+          return "[#{date}?]" if qualifier == 'questionable'
+          return "[#{date}]" if qualifier == 'inferred'
+          date
         end
-      rescue
-        date_field.content
       end
-      date_field
     end
 
-    def process_iso8601_date(date_field)
-      date_field = date_field.clone
-      date_field.content = begin
-        Date.iso8601(date_field.text).strftime('%B %d, %Y')
-      rescue
-        date_field.content
+    def parse_dates(elements)
+      # convert to DateValue objects and keep only valid ones
+      dates = elements.map { |element| DateValue.new(element) }.select(&:valid?)
+
+      # join any date ranges into DateRange objects
+      point, nonpoint = dates.partition { |date| date.point }
+      if point.any?
+        range = DateRange.new(start: point.find { |date| date.point == 'start' },
+                              stop: point.find { |date| date.point == 'end' })
+        nonpoint.unshift(range)
       end
-      date_field
-    end
+      dates = nonpoint
+
+      # ensure dates are unique with respect to their base values
+      dates = dates.group_by(&:base_value).map do |_value, group|
+        group.first if group.one?
+
+        # if one of the duplicates wasn't encoded, use that one. see:
+        # https://consul.stanford.edu/display/chimera/MODS+display+rules#MODSdisplayrules-3b.%3CoriginInfo%3E
+        if group.reject(&:encoding).any?
+          group.reject(&:encoding).first
 
-    def dedup_dates(date_fields)
-      date_text = date_fields.map { |d| normalize_date(d.text) }
-      if date_text != date_text.uniq
-        if date_fields.find { |d| d.attributes['qualifier'].respond_to?(:value) }
-          [date_fields.find { |d| d.attributes['qualifier'].respond_to?(:value) }]
-        elsif date_fields.find { |d| !d.attributes['encoding'] }
-          [date_fields.find { |d| !d.attributes['encoding'] }]
+        # otherwise just randomly pick the first in the group
         else
-          [date_fields.first]
+          group.first
         end
-      else
-        date_fields
       end
-    end
 
-    def normalize_date(date)
-      date.strip.gsub(/^\[*ca\.\s*|c|\[|\]|\?/, '')
+      # if any single dates are already part of a range, discard them
+      range_base_values = dates.select { |date| date.is_a?(DateRange) }
+                               .map(&:base_values).flatten
+      dates = dates.reject { |date| range_base_values.include?(date.base_value) }
```

Comment:
this was the missing piece to handle situations where you have e.g.
```xml
          <mods xmlns="http://www.loc.gov/mods/v3">
            <originInfo>
              <dateIssued>[1820]</dateIssued>
              <dateIssued point="start">1820</dateIssued>
              <dateIssued point="end">1825</dateIssued>
            </originInfo>
          </mods>
```

which we did already have in a test, and we wanted to output only `1820-1825`